test/runtime/document_test.cc

```diff
@@ -72,6 +72,19 @@ describe("Document", [&]() {
         "(array (true) (false))");
     });
 
+    it("handles truncated UTF16 data", [&]() {
+      char *content = reinterpret_cast<char*>(malloc(1));
+
+      spy_input->content = string((const char *)content, 1);
```

Comment:
I think that rather than allocating the content with `malloc`, you could just pass a string literal to the `std::string` constructor.

```c++
spy_input->content = string("x", 1);
```

This should behave identically with regard to triggering the bug, because either way, the `string` will copy 1 byte from the given content into its own internal buffer.

Plus, this way will avoid reading from uninitialized memory, which I would think would trigger an undefined behavior warning in its own right.